### PR TITLE
Add better error handling/output for tracking email parsing

### PR DESCRIPTION
**Pull request by CydeWeys**

### CODE REVIEW

File: lib/email_tracking_retriever.py [hunk 1]
```diff
@@ -111,37 +136,50 @@ def get_tracking_number_from_email(self,
...
     msg = email.message_from_string(str(data[0][1], 'utf-8'))
     return str(msg['To']).replace('<', '').replace('>', '')
 
+
   @retry(
       stop=stop_after_attempt(3),
       wait=wait_exponential(multiplier=1, min=2, max=16),
       reraise=True)
-  def get_tracking(self, email_id, mail) -> Tracking:
+  def get_tracking(self, email_id, mail) -> Tuple[bool, Tracking]:
```
Comment (gbrodman) [inline lib/email_tracking_retriever.py:176]: someone has been looking at golang, eh?

File: lib/email_tracking_retriever.py [hunk 2]
```diff
@@ -153,33 +191,34 @@ def get_tracking(self, email_id, mail) -> Tracking:
...
+        f"Tracking: {tracking_number}, Order(s): {order_ids}, "
+        f"Group: {group}, Status: {shipping_status}"
     )
-    if tracking_number == None:
-      tqdm.write(
-          f"Could not find tracking number from email with order(s) {order_ids}"
-      )
+    if tracking_number is None:
+      tqdm.write(f"Could not find tracking number from email; we got: {tracking}")
       self.mark_as_unread(email_id)
```
Comment (gbrodman) [inline lib/email_tracking_retriever.py:202]: should we just do the mark_as_unread step above whenever we've found that get_tracking returned False? It'd mean we don't have to have it in multiple places here

File: lib/email_tracking_retriever.py (hunk 1 above)
Comment (CydeWeys) [reply lib/email_tracking_retriever.py:162]: I haven't but this particular pattern makes so much sense that I'm essentially inventing it independently.

Also that's not true, this is a common pattern in older languages too, like C.

Python makes it easy to do with multiple return values though.

File: lib/email_tracking_retriever.py (hunk 2 above)
Comment (CydeWeys) [reply lib/email_tracking_retriever.py]: Done.

**Pull request merged**